src/server_manager/web_app/app.ts

```diff
@@ -994,7 +861,7 @@ export class App {
     this.selectedServer.addAccessKey()
         .then((serverAccessKey: server.AccessKey) => {
           const uiAccessKey = this.convertToUiAccessKey(serverAccessKey);
-          this.appRoot.getServerView(this.appRoot.selectedServer.id).addAccessKey(uiAccessKey);
+          this.appRoot.getServerView(this.appRoot.selectedServerId).addAccessKey(uiAccessKey);
```

Comment:
It seems like a bit of a code smell to use the `this.appRoot.selectedServerId` instead of the `selectedServer` in this class.

Maybe that's something we can change in a follow up PR.

Reply:
We had a selected server variable before, but it was effectively just a mirror of the one in approot, so I got rid of it (less state to keep consistent). And I really only need the ID in the UI.

Yeah, we can rethink this, especially if we merge app and app-root.